From d4d11fa693691efc11a4c74d03a60ad47db5c908 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zolt=C3=A1n=20B=C3=B6sz=C3=B6rm=C3=A9nyi?=
 <zboszor@gmail.com>
Date: Wed, 7 Jan 2026 09:22:07 +0100
Subject: [PATCH 02/18] Use a pre-built spdlog dependency
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Use <fmt/core.h> instead of <spdlog/fmt/bundled/core.h>.
Use <fmt/format.h> instead of <spdlog/fmt/bundled/format.h>
Usually spdlog does not bundle fmt.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>
Upstream-Status: Inappropriate [oe specific]
---
 hailort/cmake/external/spdlog.cmake           | 21 +------------------
 hailort/common/utils.hpp                      |  2 +-
 .../src/utils/measurement_utils.hpp           |  2 +-
 .../src/vdma/channel/channel_id.hpp           |  2 +-
 4 files changed, 4 insertions(+), 23 deletions(-)

diff --git a/hailort/cmake/external/spdlog.cmake b/hailort/cmake/external/spdlog.cmake
index 14a0949..1d4aeb1 100644
--- a/hailort/cmake/external/spdlog.cmake
+++ b/hailort/cmake/external/spdlog.cmake
@@ -1,22 +1,3 @@
 cmake_minimum_required(VERSION 3.11.0)
 
-include(FetchContent)
-
-FetchContent_Declare(
-    spdlog
-    GIT_REPOSITORY https://github.com/gabime/spdlog
-    GIT_TAG 27cb4c76708608465c413f6d0e6b8d99a4d84302 # version 1.14.1
-    GIT_SHALLOW TRUE
-    SOURCE_DIR ${HAILO_EXTERNAL_DIR}/spdlog-src
-    SUBBUILD_DIR ${HAILO_EXTERNAL_DIR}/spdlog-subbuild
-)
-
-# https://stackoverflow.com/questions/65527126/disable-install-for-fetchcontent
-FetchContent_GetProperties(spdlog)
-if(NOT spdlog_POPULATED)
-    FetchContent_Populate(spdlog)
-    if (NOT HAILO_EXTERNALS_EXCLUDE_TARGETS)
-        add_subdirectory(${spdlog_SOURCE_DIR} ${spdlog_BINARY_DIR} EXCLUDE_FROM_ALL)
-        set_target_properties(spdlog PROPERTIES POSITION_INDEPENDENT_CODE ON)
-    endif()
-endif()
+find_package(spdlog REQUIRED CONFIG)
diff --git a/hailort/common/utils.hpp b/hailort/common/utils.hpp
index b49d374..367feda 100644
--- a/hailort/common/utils.hpp
+++ b/hailort/common/utils.hpp
@@ -18,7 +18,7 @@
 
 #include "common/file_utils.hpp"
 #include "common/logger_macros.hpp"
-#include <spdlog/fmt/bundled/core.h>
+#include <fmt/core.h>
 
 #define XXH_INLINE_ALL 1
 #include "xxhash.h"
diff --git a/hailort/libhailort/src/utils/measurement_utils.hpp b/hailort/libhailort/src/utils/measurement_utils.hpp
index f677417..d667df8 100644
--- a/hailort/libhailort/src/utils/measurement_utils.hpp
+++ b/hailort/libhailort/src/utils/measurement_utils.hpp
@@ -87,7 +87,7 @@
 #define _HAILO_MEASUREMENT_UTILS_HPP_
 
 #include "measurement_utils_internal.hpp"
-#include <spdlog/fmt/bundled/format.h>
+#include <fmt/format.h>
 
 namespace hailort {
 namespace utils {
diff --git a/hailort/libhailort/src/vdma/channel/channel_id.hpp b/hailort/libhailort/src/vdma/channel/channel_id.hpp
index e9c8de2..53cbc72 100644
--- a/hailort/libhailort/src/vdma/channel/channel_id.hpp
+++ b/hailort/libhailort/src/vdma/channel/channel_id.hpp
@@ -12,7 +12,7 @@
 
 #include "hailo/hailort.h"
 #include "common/logger_macros.hpp"
-#include <spdlog/fmt/bundled/format.h>
+#include <fmt/format.h>
 #include <sstream>
 
 
-- 
2.55.0

